src/litserve/specs/openai.py

```diff
@@ -130,68 +134,88 @@ def encode_response(self, output: Union[Dict[str, str], List[Dict[str, str]]]) -
 
-    async def chat_completion(
-        self, request: ChatCompletionRequest, background_tasks: BackgroundTasks
-    ) -> ChatCompletionResponse:
+    async def chat_completion(self, request: ChatCompletionRequest, background_tasks: BackgroundTasks):
         logger.debug("Received chat completion request %s", request)
-        if request.stream:
-            raise HTTPException(400, "Stream is not supported")
 
         uids = [uuid.uuid4() for _ in range(request.n)]
         pipes = []
         for uid in uids:
             read, write = self._server.new_pipe()
-
             request_el = request.model_copy()
             request_el.n = 1
             self._server.request_buffer[uid] = (request_el, write)
             self._server.request_queue.put(uid)
-
             background_tasks.add_task(self._server.cleanup_request, self._server.request_buffer, uid)
             pipes.append((read, write))
 
-        responses = await self.get_from_pipe(uids, pipes)
-
+        responses = await self.get_from_pipes(uids, pipes)
         for read, write in pipes:
             self._server.dispose_pipe(read, write)
 
-        usage = UsageInfo()
+        if request.stream:
+            return StreamingResponse(self.streaming_completion(request, responses), media_type="application/x-ndjson")
 
-        choices = []
-        for i, data in enumerate(responses):
-            response, status = data
-            logger.debug("Received chat completion response %s with status %s", response, status)
-            if status == LitAPIStatus.ERROR:
-                load_and_raise(response)
+        return await self.non_streaming_completion(request, responses)
 
-            if status != LitAPIStatus.OK:
-                break
+    async def streaming_completion(self, request: ChatCompletionRequest, pipe_responses: List):
+        model = request.model
+        usage = None
+        async for streaming_response in azip(*pipe_responses):
+            choices = []
+            for i, chat_msg in enumerate(streaming_response):
+                chat_msg = json.loads(chat_msg)
+                logger.debug(chat_msg)
+                chat_msg = ChoiceDelta(**chat_msg)
+                choice = ChatCompletionStreamingChoice(
+                    index=i, delta=chat_msg, system_fingerprint="", usage=usage, finish_reason=None
+                )
+
+                choices.append(choice)
 
-            response = response.model_copy(update={"index": i})
-            choices.append(response)
+            chunk = ChatCompletionChunk(model=model, choices=choices, usage=usage).json()
+            logger.debug(chunk)
+            yield f"data: {chunk}\n\n"
+        yield "data: [DONE]\n\n"
 
+    async def non_streaming_completion(self, request: ChatCompletionRequest, pipe_responses: List):
         model = request.model
+        usage = UsageInfo()
+        choices = []
+        for i, streaming_response in enumerate(pipe_responses):
+            msgs = ""
```

Comment:
this should be a list, just append content here and then obtain a single string with `" ".join(msgs)`